sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -212,6 +213,8 @@ def push_snapshots(self, snapshots: t.Iterable[Snapshot]) -> None:
 
         if snapshots:
             self._push_snapshots(snapshots)
+            for snapshot in snapshots:
+                self._snapshot_cache.put(snapshot)
```

Comment:
is this going to reload the optimized query cache even if it's not necessary? we should make sure with_optimized_query only loads when necessary

Reply:
Wdym by "reload"? Reading from the file system?